compliance/evidence.py

```diff
@@ -773,26 +834,39 @@ def _store_wrapper(self, evidence_path, func, type_name):
     self.locker.add_evidence(evidence)
 
 
-def _with_evidence_decorator(evidence_paths, f, type_str):
+def _with_evidence_decorator(from_evidences, f, type_str):
     prefix = f'{type_str}/'
-    paths = map(
-        lambda p: (prefix + p) if not p.startswith(prefix) else p,
-        evidence_paths
-    )
+    from_evs = []
+    for from_ev in from_evidences:
+        path = from_ev
+        ev_class = None
+        if isinstance(from_ev, LazyLoader):
+            path = from_ev.path
+            if issubclass(from_ev.ev_class, get_evidence_class(type_str)):
```

Comment:
I think this could use an explanatory comment - if I'm reading it right, it'll make an instance of the class defined in the lazy loader if that class is a subclass of the type_str, otherwise will make the evidence be of the type in type_str.

How about:
```suggestion
        ev_class = get_evidence_class(type_str)
        if isinstance(from_ev, LazyLoader):
            path = from_ev.path
            if issubclass(from_ev.ev_class, get_evidence_class(type_str)):
                ev_class = from_ev.ev_class
```            
and save an if.

Reply:
Done in https://github.com/ComplianceAsCode/auditree-framework/pull/80/commits/a96efa7e0fa97087769f150f94373b6ff1ac3b7b